### PR TITLE
Added MSRV check

### PR DESCRIPTION
**Pull request by Razican**
This Pull Request is a follow-up of #3290.

It adds a test for the MSRV to the CI, so that we don't get similar issues in the future.

### CODE REVIEW

**Comment by jedel1043** [comment]
Can we read the root `Cargo.toml` from CI? it would be ideal to check using the provided `rust_version` from our config.

**Comment by nekevss** [comment]
Second the idea regarding reading the version in the `Cargo.toml` if possible.

**Comment by Razican** [comment]
Will see how to do this this weekend :)

**Comment by Razican** [comment]
@jedel1043, @nekevss this should be ready now :)

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -66,6 +66,24 @@ jobs:
...
+  msrv:
+    name: Minimum supported Rust version
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v4
+      # Get the rust_version from the Cargo.toml
+      - name: Get rust_version
+        id: rust_version
+        run: echo "::set-output name=rust_version::$(grep '^rust-version' Cargo.toml | cut -d' ' -f3 | tr -d '"')"
```
Comment (nekevss) [inline .github/workflows/rust.yml:78]: Huh, looks like `set-output` is going to be [deprecated soon](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/), so maybe something more akin to below.

```suggestion
        run: echo "rust_version=$(grep '^rust-version' Cargo.toml | cut -d' ' -f3 | tr -d '"')" >> $GITHUB_OUTPUT
```

**Review by nekevss — COMMENTED** [review]
Was double checking the run and noticed a deprecation warning. Maybe adjust it to use `$GITHUB_OUTPUT`?

**Review by jedel1043 — APPROVED** [review]
Nice improvement!

**Review by HalidOdat — APPROVED** [review]
Looks perfect to me! :)

**Review by nekevss — APPROVED** [review]
LGTM

**Pull request merged**